cmd/generic-handlers.go

```diff
@@ -791,14 +793,19 @@ func (h criticalErrorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 	h.handler.ServeHTTP(w, r)
 }
 
-func setSSETLSHandler(h http.Handler) http.Handler { return sseTLSHandler{h} }
+func setSSETLSHandler(h http.Handler) http.Handler {
+	if globalIsTLS {
+		return h
+	}
+	return sseTLSHandler{h}
+}
 
 // sseTLSHandler enforces certain rules for SSE requests which are made / must be made over TLS.
 type sseTLSHandler struct{ handler http.Handler }
 
 func (h sseTLSHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Deny SSE-C requests if not made over TLS
-	if !globalIsTLS && (crypto.SSEC.IsRequested(r.Header) || crypto.SSECopy.IsRequested(r.Header)) {
+	if crypto.SSEC.IsRequested(r.Header) || crypto.SSECopy.IsRequested(r.Header) {
```

Comment:
1) This check is so fast I would just leave it as-is.
2) It changes behavior AFAICT. Now `globalIsTLS` will always be true, meaning that this would never be hit in the old code. Before it would only be triggered when `globalIsTLS` is false. This does the opposite. This entire block could be removed. 

Reverting this change fixes the problem and keeps a defensive check.